src/openloco/graphics/gfx.cpp

```diff
@@ -16,7 +18,39 @@ namespace openloco::gfx
     // 0x0044733C
     void load_g1()
     {
-        call(0x0044733C);
+        auto g1Path = environment::get_path(environment::path_id::g1);
+
+        FILE * file;
+        gfx::loco_g1_header header;
+        void * g1Buffer;
+
+        file = fopen(g1Path.make_preferred().u8string().c_str(), "rb");
+
+        if (file != NULL) {
+            if (fread(&header, 8, 1, file) == 1) {
+
+                // Read element headers
+                fread(_g1Elements, header.num_entries * sizeof(loco_g1_element), 1, file);
+
+                // Read element data
+                g1Buffer = malloc(header.total_size);
+                fread(g1Buffer, header.total_size, 1, file);
```

Comment:
This needs return value checked. Since you already throw in case there is no `g1.DAT`, you could maybe add a `try { … } catch` block here, upon exception close the file and re-throw?

Reply:
Hmm... good point - I'll do just that I think.